Service/src/routes/stock.py

```diff
@@ -13,127 +13,136 @@
-		return make_response(jsonify(InvalidTicker=errors['unsupportedTicker']), 400)
+	try:
+		stockInformation = getStockInformation(ticker)
+	except requests.HTTPError as e:
+		return handleIexError(e)
 
 	return json.dumps(stockInformation)
 
 def getStockInformation(ticker):
-	requestUrl = IEX_URL_PREFIX + ticker + '/company'
+	requestUrl = f'{IEX_URL_PREFIX}{ticker}/company?token={getIexToken()}'
 
 	g.log.info('IEX API hitting: ' + requestUrl)
 	startTime = time.time()
 	rawResponse = requests.get(requestUrl)
 	iexTime = time.time() - startTime
 
-	try:
-		response = rawResponse.json()
-	except:
-		return None
+	if (rawResponse.status_code != 200):
+		rawResponse.raise_for_status()
+
+	response = rawResponse.json()
 
 	parseTime = time.time() - startTime
 	g.log.info('IEX time is: ' + str(iexTime))
 	g.log.info('Parsing data time is: ' + str(parseTime))
 
 	return response
 
-@stock_api.route('/api/v1.0/stock/<ticker>/charts', methods=['GET'])
+@stock_api.route('/api/v1.0/stock/<ticker>/chart', methods=['GET'])
 @auth.login_required
 @validator.validate_params(charts_schema.fields)
 def extract_args(ticker):
-   length = request.args.get('length')
+	length = request.args.get('length')
 
-   return get_history(ticker, length)
+	try:
+		history = get_history(ticker, length)
+	except requests.HTTPError as e:
+		return handleIexError(e)
+
+	return json.dumps(history)
 
 
 def get_history(ticker, length):
-   interval = getInterval(length)
-   requestUrl = IEX_URL_PREFIX + ticker + '/chart/' + interval
-   
-   g.log.info('IEX API hitting: ' + requestUrl)
-   startTime = time.time()
-   rawResponse = requests.get(requestUrl)
-   iexTime = time.time() - startTime
-
-   try:
-      response = rawResponse.json()
-   except:
-      return make_response(jsonify(InvalidTicker=errors['unsupportedTicker']), 400)
-
-   data = formatData(response, interval)
-   if len(data) == 0:
-      return make_response(jsonify(IEXUnavailable=errors['iexUnavailable']), 500)
-   
-   if length == 'recent':
-      data = [data[-1]]
-   
-   parseTime = time.time() - startTime
-   g.log.info('IEX time is: ' + str(iexTime))
-   g.log.info('Parsing data time is: ' + str(parseTime))
-
-   return json.dumps(data)
+	interval = getInterval(length)
+	requestUrl = f'{IEX_URL_PREFIX}{ticker}/chart/{interval}?token={getIexToken()}'
+	
+	g.log.info('IEX API hitting: ' + requestUrl)
+	startTime = time.time()
+	rawResponse = requests.get(requestUrl)
+	iexTime = time.time() - startTime
+
+	if (rawResponse.status_code != 200):
+		rawResponse.raise_for_status()
+
+	response = rawResponse.json()
+
+	data = formatData(response, interval)
+
+	if length == 'recent':
+		data = [data[-1]]
+	
+	parseTime = time.time() - startTime
+	g.log.info('IEX time is: ' + str(iexTime))
+	g.log.info('Parsing data time is: ' + str(parseTime))
 
+	return data
 
 def getSharePrice(ticker):
-   return json.loads(get_history(ticker, 'recent'))[0]['price']
+	return get_history(ticker, 'recent')[0]['price']
```

Comment:
yo there's some pretty big indent mess ups around here, do you wanna check your editor or something?

Reply:
It's my editor trying to enforce pep8 lol